ivy_tests/test_ivy/helpers.py

```diff
@@ -1804,7 +1848,32 @@ def test_frontend_function(
 
     # frontend function
     frontend_fn = ivy.functional.frontends.__dict__[frontend].__dict__[fn_tree]
-
+    
+    # check and replace NativeClass object in arguments with ivy counterparts
+    if frontend == "jax":
```

Comment:
this can all be simplified, something like:

```
convs = {"jax": convjax, "numpy": convjax, "tensorflow": convjax, "torch": convjax}
if frontend in convs:
    conv =  convs[frontend]
    args = ivy.nested_map(args, fn=conv, include_derived=True)
    kwargs = ivy.nested_map(kwargs, fn=conv, include_derived=True)
```